File: solutions/c#/ATM/CashDispenser.cs
using System;

namespace ATM
{
    public class CashDispenser
    {
        private int cashAvailable;

        public CashDispenser(int initialCash)
        {
            cashAvailable = initialCash;
        }

        public void DispenseCash(int amount)
        {
            lock (this)
            {
                if (amount > cashAvailable)
                {
                    throw new ArgumentException("Insufficient cash available in the ATM.");
                }
                cashAvailable -= amount;
                Console.WriteLine("Cash dispensed: " + amount);
            }
        }
    }
}

File: solutions/java/src/atm/CashDispenser.java
package atm;

public class CashDispenser {
    private int cashAvailable;

    public CashDispenser(int initialCash) {
        this.cashAvailable = initialCash;
    }

    public synchronized void dispenseCash(int amount) {
        if (amount > cashAvailable) {
            throw new IllegalArgumentException("Insufficient cash available in the ATM.");
        }
        cashAvailable -= amount;
        System.out.println("Cash dispensed: " + amount);
    }
}
